### PR TITLE
Optional challenge mode for Nuke Ops

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
Inspired by HBL's "meta ops"

Syndicate nuke team leaders now have an item that allows them to announce their impending assault on the station, making their shuttle unable to leave for 20 minutes in exchange for a massive telecrystal bonus (the fluff being that by showing off like this you attract more support from various syndicate benefactors).

The item has the following restrictions

-Can't be used if its more than 5 minutes into the round
-Can't be used if you've already left the base
-Can't be used with less than 50 players online

tl;dr the crew gets 20 minutes to prepare for ops but the ops get a ton of extra gear.

Still need to test everything/think over the exact timers and telecrystal bonuses

:cl: Kor
rscadd: Nuke Op team leaders can now activate challenge mode if more than 50 players are online.
/:cl:

### CODE REVIEW

**Comment by Bawhoppen** [comment]
Make it be more like 8 minutes so the Ops can discuss it before hand, 5 minutes is just when all the people who didn't want to be ops get a new ghost in there body, and people come back from the bathroom, etc.

**Comment by KorPhaeron** [comment]
@Bawhoppen 

Maybe, but that gives the ops the option of waiting even longer to deny the crew prep time. 

Ideally it'd be a 20 minute delay on their attack from the moment they use the item, but I'm afraid the crew will just call the shuttle and bail if they use it too late.

Maybe I can push back the shuttle refuel or something

**Comment by Bawhoppen** [comment]
I would argue it be giving more time to prepare, engine would already be set up, R&D started, xenobio, kitchen, medbay, chem, all started like people would normally.

**Comment by KorPhaeron** [comment]
Well no because right now the syndies can launch at 20 minutes, and if I bumped the window to use the challenge up to 8 minutes, that only leaves the crew with 12 minutes of knowing the ops are coming (which I guess is still substantial)

**Comment by Bawhoppen** [comment]
I misunderstood, I thought you meant 20 minutes after they made their statement.
But even so I don't think it matters, since theres only so much you can do to prepare.

**Comment by WJohn** [comment]
Neat idea. Adds a nice different option to blitzkrieg. I mean if they're gonna take 20 mins and be found out immediately anyway, why not just get 200 crystals extra to help?

**Comment by GunHog** [comment]
>I misunderstood, I thought you meant 20 minutes after they made their statement.
But even so I don't think it matters, since theres only so much you can do to prepare.

I must disagree here. HBL recently ran a round with the game mode set to "nuclear" instead of "secret". The Ops were given pulse carbines and a free Marauder. The crew on the other hand, were fully prepared - they turned cargo into a wall + sec barrier + reflector fortress, and a guard was placed on the AI's sat to defend against Syndieborgs.

The Ops were entirely demolished, the mech spaced, the borg crushed, and the rest of the ops were killed by their own pulse weapons. They did not get close to the Captain.

HBL taught me this lesson. <b>Do not underestimate the power of meta.</b> That being said, this has enormous potential if done properly. *This will be extremely difficult to balance, be warned.*

I wish you luck on getting this to a good spot!

**Comment by Bawhoppen** [comment]
What? HBL did an event? surely this is false propaganda 

But I think it should be 20 minutes flat after they make the announcement. Its unlikely the station will get worse after after waiting 8 minutes, it would possibly be more opportune for ops to do it earlier, so they have even less time to prepare.

**Comment by Shapsy** [comment]
Make it instantly blow up all RCD's and RPD's or you're just gonna get people hiding the disk behind 20 layers of walls (hell that'll still probably happen anyways)

**Comment by CPTANT** [comment]
Nuke ops have access to plenty of explosives to deal with such a situation.

**Comment by Shapsy** [comment]
All the C4 in the world won't get you through 20 minutes of dedicated wall and airlock spam

**Comment by Bawhoppen** [comment]
They start with actual bombs. And its not like its necessary to do this challenge-mode. Making airlocks as strong as the walls around them would also be a step in the right direction.

Also may I suggest that they unlock the ability for other special equipment in this mode? Even if they have 500 TC, theres only so much they can carry or need of the current stuff.

**Comment by Robustin** [comment]
Fun idea, probably end up shit once the meta settles (see: wall-spam and the entire station being lubed).

**Comment by Jalleo** [comment]
Teleporter does not work from centcom.
This can work but really it should announce once their shuttle is moving and give the syndies a time delay of 5 ish minuites or so. Announcing before they are even ready is silly.

**Comment by KorPhaeron** [comment]
>As it is its definitely possible to quickly set up and then get your +200TC and then teleport onto the station before the crew has finished reading the declaration.

Good point. I'm planning on replacing the teleporter soon anyway though.

**Comment by Robustin** [comment]
With penetrator rounds being proposed I double-down my support so we can get 4-5 snipers with thermals in space instakilling anyone within 2 screens of space.

Realistically that level of coordination won't happen (cue 2 guys getting lost leaving the shuttle, one guy forgetting to bring his internals and the secret macrobomb he bought killing the remaining two operatives), but damnit I want to dream.

**Comment by Bawhoppen** [comment]
The round we just did of this was fucking 10/10 amazing holy fuck best make official gamemode holy christ add, i knew HBL still had it in him

Maybe a bit more TC for ops and a more obvious game message

**Comment by KorPhaeron** [comment]
@RemieRichards @Razharas @Cheridan @phil235 @Jordie0608 @duncathan 

This has been tested on server, worked great, everyone had fun. Bug free for once.

Please merge.

**Comment by duncathan** [comment]
>bug free

hahaha as if
